src/clients/python/experimental_api_v2/library/httpclient.py

```diff
@@ -1033,6 +1033,7 @@ def infer(self,
                               request_body=request_body,
                               headers=headers,
                               query_params=query_params)
+        _raise_if_error(response)
```

Comment:
This will not work for the case output is in binary-data. We must call _raise_if_error on the json part parsed by InferResult class.

Reply:
For HTTP, this checks on the status code returned. Are you saying that this is the case where the status code will returned OK but the response is malformed?

Reply:
You are right.

Reply:
Similar changes necessary in some test scripts as well. https://github.com/NVIDIA/triton-inference-server/blob/v2test/qa/common/infer_util.py#L337
I will make them in a different PR once this is merged

Reply:
Yes in the case of inference the server always accompanies BADREQ with and error json response